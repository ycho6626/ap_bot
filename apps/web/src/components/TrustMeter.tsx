import { cn, getTrustScoreColor } from '@/lib/utils';

interface TrustMeterProps {
  score: number;
  className?: string;
}

export function TrustMeter({ score, className }: TrustMeterProps) {
  const percentage = Math.round(score * 100);
  const colorClass = getTrustScoreColor(score);

  return (
    <div className={cn('trust-bar', className)}>
      <div
        className={cn('trust-fill', colorClass)}
        style={{ width: `${percentage}%` }}
        role='progressbar'
        aria-valuenow={percentage}
        aria-valuemin={0}
        aria-valuemax={100}
        aria-label={`Trust score: ${percentage}%`}
      />
    </div>
  );
}
