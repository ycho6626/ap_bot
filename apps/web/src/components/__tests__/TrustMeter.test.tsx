import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { TrustMeter } from '../TrustMeter';

describe('TrustMeter', () => {
  it('should render trust meter with correct percentage', () => {
    render(<TrustMeter score={0.85} />);
    
    const progressBar = screen.getByRole('progressbar');
    expect(progressBar).toHaveAttribute('aria-valuenow', '85');
    expect(progressBar).toHaveAttribute('aria-valuemin', '0');
    expect(progressBar).toHaveAttribute('aria-valuemax', '100');
    expect(progressBar).toHaveAttribute('aria-label', 'Trust score: 85%');
  });

  it('should render with correct width style', () => {
    render(<TrustMeter score={0.75} />);
    
    const progressBar = screen.getByRole('progressbar');
    expect(progressBar).toHaveStyle('width: 75%');
  });

  it('should handle edge cases', () => {
    const { rerender } = render(<TrustMeter score={0} />);
    expect(screen.getByRole('progressbar')).toHaveAttribute('aria-valuenow', '0');
    
    rerender(<TrustMeter score={1} />);
    expect(screen.getByRole('progressbar')).toHaveAttribute('aria-valuenow', '100');
    
    rerender(<TrustMeter score={0.999} />);
    expect(screen.getByRole('progressbar')).toHaveAttribute('aria-valuenow', '100');
  });

  it('should apply custom className', () => {
    const { container } = render(<TrustMeter score={0.5} className="custom-class" />);
    
    expect(container.firstChild).toHaveClass('custom-class');
  });

  it('should round scores correctly', () => {
    render(<TrustMeter score={0.854} />);
    expect(screen.getByRole('progressbar')).toHaveAttribute('aria-valuenow', '85');
  });
});
