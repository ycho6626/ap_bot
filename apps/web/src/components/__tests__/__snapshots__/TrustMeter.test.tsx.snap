// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrustMeter > matches snapshot for medium trust 1`] = `
<div
  class="trust-bar"
>
  <div
    aria-label="Trust score: 72%"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="72"
    class="trust-fill bg-warning-500"
    role="progressbar"
    style="width: 72%;"
  />
</div>
`;
